src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/rc_helpers_test.cc

```diff
@@ -0,0 +1,278 @@
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  EXPECT_NE(result, data1);
+  EXPECT_EQ(result, data2);
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_FirstObjectDontContainKey) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Map);
+
+  std::map<std::string, std::string> expected_keys_values = {
+      {"key1", "value1"}, {"key2", "value2"}};
+  auto it = expected_keys_values.begin();
+
+  data1[it->first] = it->second;
+  ++it;
+  data2[it->first] = it->second;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  for (auto& e : expected_keys_values) {
+    EXPECT_TRUE(result.keyExists(e.first));
+    EXPECT_EQ(result[e.first], expected_keys_values[e.first]);
+  }
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_SecondObjectContainsDifferentValueType) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Map);
+
+  data1["key1"] = "value1";
+  data2["key1"] = 30;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  EXPECT_EQ(data2["key1"], result["key1"]);
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_Recursive) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Map);
+
+  data1["key1"]["subkey1"] = "subvalue1";
+  data2["key1"]["subkey2"] = 30;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  EXPECT_EQ(result["key1"]["subkey1"], data1["key1"]["subkey1"]);
+  EXPECT_EQ(result["key1"]["subkey2"], data2["key1"]["subkey2"]);
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_MergeObjectsTharContainArrays) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data1_array(SmartType_Array);
+  (*data1_array.asArray()) =
+      SmartArray{genSmartObject(1, true), genSmartObject(2, false)};
+  data1[kArray] = data1_array;
+
+  SmartObject data2(SmartType_Map);
+  SmartObject data2_array(SmartType_Array);
+  (*data2_array.asArray()) =
+      SmartArray{genSmartObject(3, true), genSmartObject(4, false)};
+  data2[kArray] = data2_array;
+
+  SmartObject expected_result(SmartType_Map);
+  SmartObject expected_result_array(SmartType_Array);
+  (*expected_result_array.asArray()) =
+      SmartArray{data2[kArray].getElement(0), data2[kArray].getElement(1)};
+
+  expected_result[kArray] = expected_result_array;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  EXPECT_EQ(expected_result, result);
+}
+
+TEST_F(RCHelpersTest, MergeArray_OneOfParamsNotSmartArray) {
+  using namespace smart_objects;
+  SmartObject data1(SmartType_Array);
+  SmartObject data2(SmartType_Map);
+
+  auto result = RCHelpers::MergeArray(data1, data2);
+  EXPECT_NE(result, data1);
+  EXPECT_EQ(result, data2);
+}
+
+TEST_F(RCHelpersTest, MergeArray_FirstArrayEmpty) {
+  using namespace smart_objects;
+  SmartObject data1(SmartType_Array);
+  SmartObject data2(SmartType_Array);
+  *data2.asArray() = std::vector<SmartObject>{
+      SmartObject(10),
```

Comment:
@YarikMamykin MAgic numbers?

Reply:
@AByzhynar , fixed in b61ba7c099f2c1b7caf78e8180514ad3e9cf253a